src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/optimizer/dynamicProgramming/WrappingDPAlgorithm.java

```diff
@@ -0,0 +1,63 @@
+package se.liu.ida.hefquin.engine.queryproc.impl.optimizer.dynamicProgramming;
+
+import se.liu.ida.hefquin.engine.federation.FederationAccessException;
+import se.liu.ida.hefquin.engine.queryplan.LogicalOperator;
+import se.liu.ida.hefquin.engine.queryplan.LogicalPlan;
+import se.liu.ida.hefquin.engine.queryplan.PhysicalPlan;
+import se.liu.ida.hefquin.engine.queryplan.logical.impl.LogicalOpMultiwayJoin;
+import se.liu.ida.hefquin.engine.queryplan.physical.PhysicalOperatorForLogicalOperator;
+import se.liu.ida.hefquin.engine.queryproc.QueryOptimizationException;
+import se.liu.ida.hefquin.engine.queryproc.impl.optimizer.QueryOptimizationContext;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class WrappingDPAlgorithm {
+    protected final QueryOptimizationContext ctxt;
+
+    public WrappingDPAlgorithm( final QueryOptimizationContext ctxt ) {
+        assert ctxt != null;
+        this.ctxt = ctxt;
+    }
+
+    public PhysicalPlan optimize( final LogicalPlan initialPlan ) throws FederationAccessException, QueryOptimizationException {
```

Comment:
As in the other PR, none of these public methods should declare to throw FederationAccessException.